src/main/java/org/jenkinsci/plugins/workflow/cps/view/InterpolatedSecretsAction.java

```diff
@@ -0,0 +1,164 @@
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Action to generate the UI report for watched environment variables
+ */
+@Restricted(NoExternalUse.class)
+public class InterpolatedSecretsAction implements RunAction2 {
+
+    private List<InterpolatedWarnings> interpolatedWarnings = new ArrayList<>();
+    private transient Run<?, ?> run;
+
+    public String getIconFileName() {
+        return null;
+    }
+
+    public String getDisplayName() {
+        return null;
+    }
+
+    public String getUrlName() {
+        return null;
+    }
+
+    public void record(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull String nodeId) {
+        interpolatedWarnings.add(new InterpolatedWarnings(stepName, interpolatedVariables, run, nodeId));
+    }
+
+    public List<InterpolatedWarnings> getWarnings() {
+       return interpolatedWarnings;
+    }
+
+    public boolean hasWarnings() {
+        if (interpolatedWarnings.isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public boolean isInProgress() {
+        return run.isBuilding();
+    }
+
+    @Override
+    public void onAttached(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @ExportedBean
+    public static class InterpolatedWarnings {
+        final String stepName;
+        final List<String> interpolatedVariables;
+        final Run run;
+        final String nodeId;
+
+        private InterpolatedWarnings(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull Run run, @Nonnull String nodeId) {
+            this.stepName = stepName;
+            this.interpolatedVariables = interpolatedVariables;
+            this.run = run;
+            this.nodeId = nodeId;
+        }
+
+        @Exported
+        public String getStepSignature() {
+            StringBuilder sb = new StringBuilder();
+            Map<String, Object> stepArguments;
+            try {
+                stepArguments = getStepArguments(run, nodeId);
+            } catch (IllegalStateException e) {
+                return "Unable to construct " +  stepName + ": " + e.getMessage();
+            }
+
+            sb.append(stepName + "(");
+            Set<Map.Entry<String, Object>> entrySet = stepArguments.entrySet();
+            if (!entrySet.isEmpty()) {
+                // Give some sort of order to the step arguments
+                TreeSet<String> sortedArgs = new TreeSet<>();
```

Comment:
Not sure if it's necessary to sort the arguments. I thought some order would be nice.  However, arguments in `UninstantiatedDescribable` are not sorted. Any thoughts?

Reply:
Updated `ArgumentsActionImpl.sanitizeMapAndRecordMutation()` to used `LinkedHashMap` to preserve arguments order in 163e7cb7452204f9270783c0925f0f5c0113caae